src/vm.rs

```diff
@@ -34,35 +34,34 @@ impl Vm {
         if self.instructions.len() % 2 != 0 {
             panic!("Instruction array is invalid");
         }
-        let mut i = 0;
-        while i < self.instructions.len() {
-            let operation = self.instructions.get(i).expect("Failed to get Operation");
-            let raw_argument = self
-                .instructions
-                .get(i + 1)
-                .expect("Failed to get Operation argument");
-            let argument = match raw_argument {
-                Operation::Argument(v) => *v,
+
+        for instr_pair in self.instructions.chunks_exact(2) {
+            let operation = instr_pair[0];
+            let argument = match instr_pair[1] {
```

Comment:
Grouping the instructions into operation and argument pairs can be simplified by using the chunk functions.
Allows for doing this easily without the need to self-manage a counter and stuff